rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,164 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
+	if err != nil {
+		return nil, nil, err
+	}
+	meanZ, stdevZ, err := calculateMeanAndStandardDeviation(Z)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	maxX := math.Min(meanX+float64(sigmaLevel)*stdevX, meta.MaxX)
+	minX := math.Max(meanX-float64(sigmaLevel)*stdevX, meta.MinX)
+	maxZ := math.Min(meanZ+float64(sigmaLevel)*stdevZ, meta.MaxZ)
+	minZ := math.Max(meanZ-float64(sigmaLevel)*stdevZ, meta.MinZ)
+
+	// Edits the max and min values to ensure the robot marker is in the image
+	var robotMarker spatialmath.Pose
+	if ppRM.robotPose != nil {
+		robotMarker = *ppRM.robotPose
+		maxX = math.Max(maxX, robotMarker.Point().X)
+		minX = math.Min(minX, robotMarker.Point().X)
+		maxZ = math.Max(maxZ, robotMarker.Point().Z)
+		minZ = math.Min(minZ, robotMarker.Point().Z)
+	}
+
+	// Calculate scale factor
+	widthScaleFactor := float64(imageWidth) / (maxX - minX)
+	heightScaleFactor := float64(imageHeight) / (maxZ - minZ)
+
+	// Add points to a new image
+	color := rimage.NewImage(imageWidth, imageHeight)
+	cloud.Iterate(0, 0, func(pt r3.Vector, data pointcloud.Data) bool {
+		j := (pt.X - meta.MinX) * widthScaleFactor
+		i := (pt.Z - meta.MinZ) * heightScaleFactor
+		x, y := int(math.Round(j)), int(math.Round(i))
+
+		// Adds a point to an image using the value to define the color. If no value is available,
+		// a default color of white is used.
+		if x >= 0 && x < imageWidth && y >= 0 && y < imageHeight && data != nil {
+			var c rimage.Color
+			if data.HasValue() {
+				c = getProbabilityColorFromValue(data.Value())
+			} else {
+				c = rimage.NewColor(255, 255, 255)
+			}
+			addPointToImage(color, image.Point{x, y}, c, voxelSize)
+		}
+		return true
+	})
+
+	// Add robot marker to image
+	if ppRM.robotPose != nil {
+		robotMarkerPoint := image.Point{
+			X: int(math.Round((robotMarker.Point().X - meta.MinX) * widthScaleFactor)),
+			Y: int(math.Round((robotMarker.Point().Z - meta.MinZ) * heightScaleFactor)),
+		}
+		robotMarkerColor := rimage.NewColor(255, 0, 0)
+		addPointToImage(color, robotMarkerPoint, robotMarkerColor, voxelSize*robotMarkerRatio)
+	}
+	return color, nil, nil
+}
+
+// RGBDToPointCloud calls its equivalent in ParallelProjection.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) RGBDToPointCloud(
+	img *rimage.Image,
+	dm *rimage.DepthMap,
+	crop ...image.Rectangle,
+) (pointcloud.PointCloud, error) {
+	pp := ParallelProjection{}
+	return pp.RGBDToPointCloud(img, dm, crop...)
+}
+
+// ImagePointTo3DPoint calls its equivalent in ParallelProjection.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
+	pp := ParallelProjection{}
+	return pp.ImagePointTo3DPoint(pt, d)
+}
+
+// getProbabilityColorFromValue returns the RGB color values based on the the probability value and defined thresholds
+// TODO (RSDK-1074): once probability values are available this function should be changed to produced desired images.
```

Comment:
See PR description

Reply:
Maybe make another ticket to tune this? I'm worried it will get lost as part of RSDK-1074.